extensions/markdown-language-features/src/preview/preview.ts

```diff
@@ -672,6 +672,12 @@ export class DynamicMarkdownPreview extends Disposable implements IManagedMarkdo
 		this._register(webview.onDidDispose(() => { this.dispose(); }));
 
 		this._register(this._webviewPanel.onDidChangeViewState(e => {
+			// refresh dynamic preview when it becomes inactive,
+			// this will avoid the stale content when the preview is reactivated
+			// https://github.com/microsoft/vscode/issues/165188
+			if (!e.webviewPanel.active) {
+				this.refresh();
```

Comment:
Not a fan of this approach as it results in extra rendering of the markdown 

What about storing the current markdown inside the webview using `setState`?

Reply:
>  storing the current markdown inside the webview using setState

I tried but didn't figure out how to do it. The nearest thing I camp up is updating current webiview HTML:

```js
const document = await vscode.workspace.openTextDocument(this._resource);
const content = await this._contentProvider.renderDocument(document, this, this._previewConfigurations, this._line, undefined, this.state, this._disposeCts.token);
this._webviewPanel.webview.html = content.html;
```

Could you please elaborate?